### PR TITLE
[Behat][Repositories] Refactory find by name repository methods

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes
| Related tickets | https://github.com/Sylius/Sylius/pull/6137, https://github.com/Sylius/Sylius/issues/5374
| License         | MIT

As entities names are not unique, whole concept of ``findOneByName`` repository methods is incorrect and leads to some potential errors. I've changed each ``findOneByName`` to ``findByName``, therefore giving possibility to return zero or multiple entities.
Moreover, currently these methods are used only in Behat contexts. Instead of asserting ``notNull`` in them (basically in transformer's methods), now we assets ``eq(1,...)``, to be sure we operate on **one** and only **one** object.

### CODE REVIEW

File: src/Sylius/Bundle/PaymentBundle/Doctrine/ORM/PaymentMethodRepository.php [hunk 1]
```diff
@@ -28,25 +28,10 @@ public function findByName(array $names)
...
-        return $this->createQueryBuilder('o')
-            ->addSelect('translation')
-            ->leftJoin('o.translations', 'translation')
-            ->where('translation.name = :name')
             ->setParameter('name', $name)
             ->getQuery()
-            ->getOneOrNullResult()
-        ;
+            ->getResult()
+            ;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/PaymentBundle/Doctrine/ORM/PaymentMethodRepository.php]: Should be reverted

File: src/Sylius/Behat/Context/Setup/TaxationContext.php [hunk 2]
```diff
@@ -173,12 +174,18 @@ public function theTaxRateIsOfAmount(TaxRateInterface $taxRate, $amount)
      */
     private function getOrCreateTaxCategory($taxCategoryName)
     {
-        $taxCategory = $this->taxCategoryRepository->findOneByName($taxCategoryName);
-        if (null === $taxCategory) {
-            $taxCategory = $this->createTaxCategory($taxCategoryName);
+        $taxCategories = $this->taxCategoryRepository->findByName($taxCategoryName);
+        if (empty($taxCategories)) {
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/TaxationContext.php]: I don't like this logic, but it can be case for another PR

File: src/Sylius/Behat/Context/Setup/TaxationContext.php [hunk 3]
```diff
@@ -173,12 +174,18 @@ public function theTaxRateIsOfAmount(TaxRateInterface $taxRate, $amount)
...
+            $taxCategories = [$this->createTaxCategory($taxCategoryName)];
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/TaxationContext.php]: `return $this->createTaxCategory($taxCategoryName)` 😉

File: src/Sylius/Component/Payment/Repository/PaymentMethodRepositoryInterface.php [hunk 4]
```diff
@@ -21,17 +21,10 @@
 interface PaymentMethodRepositoryInterface extends RepositoryInterface
 {
     /**
-     * @param array $names
+     * @param string $names
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Payment/Repository/PaymentMethodRepositoryInterface.php]: `name`

**Comment by michalmarcinkowski** [comment]
You should also apply [my comment](https://github.com/Sylius/Sylius/pull/6137#issuecomment-248406620), for translatable resources we should specify a locale in which we would like to search.

**Comment by Zales0123** [comment]
@michalmarcinkowski Done 👍

**Comment by michalmarcinkowski** [comment]
Thanks Mateusz! 👍

File: src/Sylius/Behat/Context/Transform/ProductContext.php [hunk 5]
```diff
@@ -41,14 +41,15 @@ public function __construct(ProductRepositoryInterface $productRepository)
      */
     public function getProductByName($productName)
     {
-        $product = $this->productRepository->findOneByName($productName);
+        $products = $this->productRepository->findByName($productName, 'en_US');
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Transform/ProductContext.php]: For now we use English language in Behat I think, but it obviously should be changed (getting locale from shared storage?)

Comment (lchrusciel) [inline src/Sylius/Behat/Context/Transform/ProductContext.php]: Passing locale as a second argument with default value?

Comment (pamil) [inline src/Sylius/Behat/Context/Transform/ProductContext.php]: In fact, we should use `%locale%` parameter, but I've got no idea how to pass it here right now.

Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Transform/ProductContext.php]: @pamil that's the reason for having it hardcoded for now 😉
